backendservice/src/main/java/com/google/gpay/virtualqueue/backendservice/repository/InMemoryVirtualQueueRepository.java

```diff
@@ -122,4 +124,25 @@ public UpdateShopStatusResponse updateShop(UpdateShopStatusRequest updateShopSta
 		shopMap.get(updateShopStatusRequest.getShopId()).setStatus(updateShopStatusRequest.getShopStatus());
 		return new UpdateShopStatusResponse(shopMap.get(updateShopStatusRequest.getShopId()));
 	}
+
+	public Shop getShopByShopId(UUID shopId) {
+		if (shopMap.get(shopId).getStatus() == ShopStatus.ACTIVE) {
+			return shopMap.get(shopId);
+		}
+		// TODO: Throw exception here.
+		Logger.getLogger(InMemoryVirtualQueueRepository.class.getName()).log(Level.SEVERE,
+				"Tried to fetch information of a shop which is not in the ACTIVE state.");
+		return new Shop();
+	}
+
+	public long getCustomersInQueue(UUID shopId) {
+		List<Token> tokenList = shopIdToListOfTokensMap.get(shopId);
+		long customersInQueue = 0L;
+		for (int i = 0; i < tokenList.size(); i++) {
+			if (tokenList.get(i).getStatus() == Status.ACTIVE) {
+				customersInQueue++;
+			}
+		}
```

Comment:
Can we use java streams here instead?

Reply:
Yes changed and changed data type of customersInQueue to AtomicInteger as could not increment long variable in lambda expression. Please advice if there is a better approach.